### PR TITLE
Gatsby Global Code Refactor

### PR DESCRIPTION
**Pull request by tiloyi**
[skip ci]

## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [ ] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [ ] No

## Describe the changes

GitHub issue number or Jira issue URL: N/A

### CODE REVIEW

**Comment by mohamedMok** [comment]
https://featgatsbycoderefactor-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com/

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ jobs:
         with:
           node-version: 18
       - run: npm install yarn -g
+      - run: yarn install
       - run: yarn lerna
```
[the remark this answers is not in the document]
Comment (tiloyi) [reply .github/workflows/push.yml:32]: Hi @mohamedMok ,
Thanks for your review.

I've corrected the problems with navigation & icons.

Concerning the IC, I'll do the fix in another PR related to the Lerna update to version 8.

File: packages/gatsby-theme-styleguide/src/components/Menu/MenuList.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    const openState = currentItem.dataset.open === 'false' ? false : true
+
+    currentItem.dataset.open = `${!openState}`
+  }
+
+  return (
+    <ul className={styles.menuList}>
+      {items.map((item, index) => {
+        const hasChildren = item.content.length > 0
+        const isOpen = item.isOpen
```
Comment (mohamedMok) [inline packages/gatsby-theme-styleguide/src/components/Menu/MenuList.js:22]: Well done 
🔥

**Review by mohamedMok — APPROVED** [review]
Well done @tiloyi 💯

**Comment by SimonCfn** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [5ec782a6\-81e3\-431d\-8db3\-296ce662bc71](https://eu.ast.checkmarx.net/projects/d2ea2837-0ed1-4404-8c6a-f1a1376ffd05/scans?id=5ec782a6-81e3-431d-8db3-296ce662bc71&branch=feat%2Fgatsby-code-refactor)
### New Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
MEDIUM|[Client_Potential_XSS](https://eu.ast.checkmarx.net/results/d2ea2837-0ed1-4404-8c6a-f1a1376ffd05/5ec782a6-81e3-431d-8db3-296ce662bc71/sast/description/79/7567754570932744334)|/packages/gatsby\-theme\-styleguide/src/components/AreaContact/AreaContact\.js: [5](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//packages/gatsby-theme-styleguide/src/components/AreaContact/AreaContact.js#L5)|[Attack Vector](https://eu.ast.checkmarx.net/results/5ec782a6-81e3-431d-8db3-296ce662bc71/d2ea2837-0ed1-4404-8c6a-f1a1376ffd05/sast?result-id=L7hFir%2F2cQH1vgKmGBSF%2BZoiwSo%3D&redirect=true)
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [46](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L46)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [40](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L40)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [31](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L31)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [43](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L43)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [53](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L53)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [58](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L58)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [37](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L37)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [52](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L52)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [65](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L65)|Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps...

### Fixed Issues
Severity|Issue|Source File / Package
---|---|---
MEDIUM|[Client_Potential_XSS](https://eu.ast.checkmarx.net/results/d2ea2837-0ed1-4404-8c6a-f1a1376ffd05/5ec782a6-81e3-431d-8db3-296ce662bc71/sast/description/79/7567754570932744334)|/packages/gatsby\-theme\-styleguide/src/components/AreaContact/AreaContact\.js: [41](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//packages/gatsby-theme-styleguide/src/components/AreaContact/AreaContact.js#L41)
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [45](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L45)
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [57](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L57)
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [42](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L42)
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [52](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L52)
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [36](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L36)
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [30](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L30)
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [64](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L64)
MEDIUM|Unpinned Actions Full Length Commit SHA|/release\.yml: [39](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/release.yml#L39)
MEDIUM|Unpinned Actions Full Length Commit SHA|/push\.yml: [51](https://github.com/adeo/mozaic-design-system/blob/feat/gatsby-code-refactor//.github/workflows/push.yml#L51)

**Pull request merged**